### DISCUSSION TITLE
What's the current state of elsa visual designer

### DISCUSSION THREAD

**Discussion by ahmed-adly-khalil** [elsa-workflows/elsa-core#1597] [Design & tech]
Hello,
I have noticed that the repo https://github.com/elsa-workflows/elsa-designer has been archived, has the visual designer work moved to https://github.com/elsa-workflows/elsa-designer-react ? 

also if i want to contribute to the visual designer project how can i do that?

Thank you
Ahmed

**Comment by sfmskywalker** [comment]
Hi @ahmed-adly-khalil ,

The visual designer code has moved into the core repo and can be found [here](https://github.com/elsa-workflows/elsa-core/tree/master/src/designer/elsa-workflows-studio).

If you want to contribute to the visual designer project, then that would be awesome!

It's best to first create an issue to describe the issue or feature you want to work on and then create a PR.
The easiest way to do that is by forking this repo, cloning your fork to your computer, and start making changes and push them.
When you think you're ready, create a PR and I will automatically be notified about that so I can take a look.

**Reply by ahmed-adly-khalil** [reply]
Awesome, I will do that. 
What about the react repo https://github.com/elsa-workflows/elsa-designer-react ? I'm more experienced with react and I think i can add more value if the visual designer has shifted from stencil to react, is there a chance that this could happen?

**Reply by sfmskywalker** [reply]
That repo is intended to host a react wrapper around the designer web component.

The reason the designer is done with Stencil is because it is framework-agnostic, allowing users to use the component as regular HTML elements and of course wrap them in Angular, React, Vue, Svelte, Blazor, etc.

Moving the implementation to React would suddenly require all these users to take a dependency on React.

Stencil is very similar to React, so things should feel quite familiar (certain details are different of course).

**Reply by ahmed-adly-khalil** [reply]
Awesome, will fork and start looking into it

**Comment by ahmed-adly-khalil** [comment]
Forked Today, Work starts tomorrow :)

**Reply by ahmed-adly-khalil** [reply]
Link here: https://github.com/Blazi-Commerce/elsa-core

**Comment by GeorgeS2019** [comment]
FYI: 
[Consider Elsa Workflows Studio in Blazor](https://github.com/elsa-workflows/elsa-core/issues/3810)
